---
output: html_document
editor_options: 
  chunk_output_type: console
---
```{r setup, echo=FALSE, message=FALSE, include=FALSE, error=FALSE}
library(GenomicRanges)
library(rtracklayer)
library(parallel)
library(ggplot2)
library(plyr)
library(doMC)
library(dplyr)
library(magrittr)
library(Hmisc)
library(cowplot)

registerDoMC(cores=10)

# Output folder for this document
options(knitr.figure_dir = "figure4_mnase_metagene_promoter_distal_muscle_0_200")

knitr::opts_chunk$set(dev=c("png", "pdf"))

source("shared/knitr_common.r")
source("shared/granges_common.r")
source("shared/metagene_common.r")
source("shared/flybase.r")


GRangesFromDataFrames<-function(dataframe){with(dataframe,GRanges(IRanges(start=start,end=end),seqnames=seqnames,strand=strand,dataframe%>%dplyr::select(-strand,-start,-end,-seqnames,-width)))}

build_read_plot<-function(region_group,mnase_lst,region_lst){
  mnase_lst %>% lapply(get_metagene_reads,region_lst[[region_group]],before_tss=500, after_tss=500, smooth=41,
                                        normalization_target=20000000*150) %>% bind_rows(.id="sample")
}


make_plot_for_each_group<-function(group,lola_peaks.gr){
lola_peaks.gr%<>%{.[.$group==group]}

lola_peaks.df<-metagene_compatible_df(lola_peaks.gr) 
region_lst<-list(lola_binding_site=lola_peaks.df)
region_lst_names<-names(region_lst) 

mclapply(setNames(region_lst_names,region_lst_names),build_read_plot,mnase_lst,region_lst,mc.cores=4) %>% bind_rows(.id="region_group") ->mnase_metagene

mnase_metagene %>% ggplot(aes(x=tss_distance, ymax=smooth,ymin=10,fill=sample)) +
       geom_ribbon(position="identity",alpha=0.4) + xlim(-500,500)+
       labs(x     = "Centered At Lola peak", 
            y     = "Mnase signal",
            title = paste0("Nucleosome Occupancy changes ",group)) + 
        theme(text=element_text(size=25))+theme_bw()+theme(
             panel.grid.major=element_blank(),
             panel.grid.minor=element_blank())
}


get_lola_peaks<-function(location="promoter",number=1000,promoter_width=500,upstream=50){

import("../bed/lola_i_peaks_1_summits.bed") %>% resize(fix='center',width=200) ->lola_peaks_1.gr   
import("../bed/lola_i_peaks_2_summits.bed") %>% resize(fix='center',width=200) ->lola_peaks_2.gr
lola_peaks_1.gr%<>%{.[order(.$score,decreasing=TRUE)][1:number]}
lola_peaks_2.gr%<>%{.[order(.$score,decreasing=TRUE)][1:number]}

lola_peaks.gr <- lola_peaks_1.gr[findOverlaps(lola_peaks_1.gr,lola_peaks_2.gr) %>% queryHits %>% unique()] %>% resize(fix='center',width=1)
valid.chrs<-c("chr2L","chr2R","chr3L","chr3R","chrX","chr4")
lola_peaks.gr%<>%keepSeqlevels(valid.chrs,pruning.mode="coarse")
lola_peaks.gr%<>%assign_nearest_gene(.)
lola_peaks.gr$location<-'distal'
lola_peaks.gr$location[flybase_txs_granges() %>% resize(fix='start',width=upstream) %>% resize(fix='end',width=promoter_width-upstream) %>% findOverlaps(lola_peaks.gr) %>% subjectHits]<-'promoter'
lola_peaks.gr%<>%.[.$location %in% location]
lola_peaks.gr
}

metagene_compatible_df<-function(region.gr){
  
region.gr %>% as.data.frame ->region.df
region.df$strand<-ifelse(region.df$strand=='+',1,-1)
region.df$chr<-region.df$seqnames
region.df  
}

filter_for_motif<-function(gr,motif,mismatch=0,ignore_strand=TRUE){
  g_match<-vmatchPattern(pattern=motif,subject=Dmelanogaster,max.mismatch=mismatch,fixed=FALSE)
  mcols(gr)$motif<-countOverlaps(gr,g_match,ignore.strand=ignore_strand)
  gr<-gr[gr$motif>=1]
  gr
}
```


Look at the pol II profile at Lola target genes

### Lola targets are defined by Lola binding- top 1000 peaks
```{r header_child, child="child_docs/header_child.Rmd"}
```

```{r files}

mnase_2_4.cov<-((import("../bw/Dme_MNase_2-4h_toll10b_1.bam_from0to200_rpm.bw") %>% {coverage(.,weight=.$score)})+(import("../bw/Dme_MNase_2-4h_toll10b_2.bam_from0to200_rpm.bw") %>% {coverage(.,weight=.$score)}))/2
                
mnase_14_17.cov<- (import("../bw/Dme_MNase_14-17h_muscle_1.bam_from0to200_rpm.bw") %>% {coverage(.,weight=.$score)})

mnase_lst<-list(mnase_2_4=mnase_2_4.cov,mnase_14_17=mnase_14_17.cov)

````


```{r plot_reads_metagene_fw,dev=c('png', 'pdf'),fig.width=12,fig.height=12}

lola_peaks_promoter.gr<-get_lola_peaks("promoter",750) %>% resize(fix='center',width=200) %>% filter_for_motif(motif="AAAGCTY")  %>% resize(fix='center',width=1)
lola_peaks_distal.gr<-get_lola_peaks("distal",750) %>% resize(fix='center',width=200) %>% filter_for_motif(motif="AAAGCTY")  %>% resize(fix='center',width=1)

lola_peaks_promoter.df<-metagene_compatible_df(lola_peaks_promoter.gr)
lola_peaks_distal.df<-metagene_compatible_df(lola_peaks_distal.gr)

region_lst<-list(lola_binding_promoter=lola_peaks_promoter.df,lola_binding_distal=lola_peaks_distal.df)
region_lst_names<-names(region_lst) 

mclapply(setNames(region_lst_names,region_lst_names),build_read_plot,mnase_lst,region_lst,mc.cores=4) %>% bind_rows(.id="region_group") ->mnase_metagene

mnase_metagene %>% ggplot(aes(x=tss_distance)) +
       geom_ribbon(aes(ymin=0,ymax=smooth,fill=sample,alpha=0.3)) + xlim(-500,500)+
  facet_wrap(~region_group,scale="free_y")+
       labs(x     = "centered At Lola peak aligned to TSS", 
            y     = "Mnase signal",
            title = paste0("Nucleosome Occupancy changes")) +
        theme(text=element_text(size=25))+theme_bw()+theme_cowplot()

mnase_metagene %>% ggplot(aes(x=tss_distance, y=smooth,color=sample)) +
       geom_line(position="identity") + xlim(-500,500)+
  facet_wrap(~region_group,scale="free_y")+
       labs(x     = "Centered At Lola peak", 
            y     = "Mnase signal",
            title = paste0("Nucleosome Occupancy changes "))+
        theme(text=element_text(size=25))+theme_bw()+theme(
             panel.grid.major=element_blank(),
             panel.grid.minor=element_blank())

#Centered at lola binding site but aligned to TSS

lola_peaks_promoter.gr<-get_lola_peaks("promoter",500) %>% resize(fix='center',width=500) %>% filter_for_motif(motif="AAAGCTY")  %>% resize(fix='center',width=1)
lola_peaks_promoter.df<-metagene_compatible_df(lola_peaks_promoter.gr)

promoter.df<- flybase_txs_granges() %>% filter_chrs %>% as.data.frame 

promoter.df %<>%
		 	 	dplyr::group_by(fb_gene_id) %>% 
		 	 	dplyr::summarise(fb_tx_id  = fb_tx_id[1]) %>%
		 	 	 `$`(fb_tx_id) %>% { subset(promoter.df, subset = fb_tx_id %in% .) }

lola_peaks_promoter.df$strand<-'1'
lola_peaks_promoter.df$strand[lola_peaks_promoter.df$nearest_gene_id %in% (promoter.df$fb_gene_id[promoter.df$strand=="-"])]<-'-1'

region_lst<-list(lola_binding_promoter=lola_peaks_promoter.df)
region_lst_names<-names(region_lst) 

mclapply(setNames(region_lst_names,region_lst_names),build_read_plot,mnase_lst,region_lst,mc.cores=4) %>% bind_rows(.id="region_group") ->mnase_metagene

mnase_metagene %>% ggplot(aes(x=tss_distance)) +
       geom_ribbon(aes(ymin=0,ymax=smooth,fill=sample,alpha=0.3)) + xlim(-500,500)+
  facet_wrap(~region_group,scale="free_y")+
       labs(x     = "centered At Lola peak aligned to TSS", 
            y     = "Mnase signal",
            title = paste0("Nucleosome Occupancy changes")) +
        theme(text=element_text(size=25))+theme_bw()+theme_cowplot()

mnase_metagene %>% ggplot(aes(x=tss_distance, y=smooth,color=sample)) +
       geom_line(position="identity") + xlim(-500,500)+
  facet_wrap(~region_group,scale="free_y")+
       labs(x     = "centered At Lola peak aligned to TSS", 
            y     = "Mnase signal",
            title = paste0("Nucleosome Occupancy changes "))+
        theme(text=element_text(size=25))+theme_bw()+theme_cowplot()

```
``` {r session_info_child, child="child_docs/session_info_child.Rmd"}
```